play28-json-ops/src/main/scala/play/api/libs/json/ops/v4/RecoverOps.scala

```diff
@@ -0,0 +1,97 @@
+package play.api.libs.json.ops.v4
+
+import play.api.libs.json._
+
+import scala.language.higherKinds
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+trait RecoverOps[F[x] <: Reads[x], A] extends Any {
```

Comment:
This is bad for forward binary compatibility. I should either fix it or resign to rewriting this in v5.

Reply:
I can't think of a better way to do this for now. I'm gonna leave this as an issue for v5 to figure out.